### PR TITLE
PHP-FPM: unified timeouts across nginx/php-fpm, TCP use instead of unix socket

### PR DESCRIPTION
**Pull request by bryanlatten**

### CODE REVIEW

**Comment by ingluisjimenez** [comment]
:+1:

File: Dockerfile [hunk 1]
```diff
@@ -48,7 +48,9 @@ RUN pecl install igbinary-1.2.1 && \
 
 # Prevent newrelic daemon from auto-spawning; uses newrelic run.d script to enable at runtime, when ENV variables are present
 # @see https://docs.newrelic.com/docs/agents/php-agent/advanced-installation/starting-php-daemon-advanced
-RUN sed -i "s/;newrelic.daemon.dont_launch = 0/newrelic.daemon.dont_launch = 3/" /etc/php5/mods-available/newrelic.ini
+RUN sed -i "s/;newrelic.daemon.dont_launch = 0/newrelic.daemon.dont_launch = 3/" /etc/php5/mods-available/newrelic.ini && \
+    sed -i "s/listen = \(.*\)\+/listen = 127.0.0.1:9000/" /etc/php5/fpm/pool.d/www.conf
```
Comment (ingluisjimenez) [inline Dockerfile:52]: :skull: unix sockets

I read last night that unix sockets can fail often during high load.

**Comment by iderdik** [comment]
LGTM

File: Dockerfile (hunk 1 above)
Comment (bryanlatten) [inline Dockerfile:52]: @ingluisjimenez sadly, the prod ones were barely under any load

**Comment by chrishenry** [comment]
Here's another timeout value I found;

```
http {
    #...
        fastcgi_read_timeout 300; 
    #...
}
```

https://rtcamp.com/tutorials/php/increase-script-execution-time/
http://nginx.org/en/docs/http/ngx_http_fastcgi_module.html#fastcgi_read_timeout

File: Dockerfile (hunk 1 above)
Comment (ingluisjimenez) [inline Dockerfile:52]: right, other than the health check, there is no load.

I was thinking if the cause might be related with the selfie checks and the requests not being terminated properly, which would leave the process running for more time, resulting on all children busy and no more of them available for nginx.

When we docker-entered the container, the load was high (above 1.00) for the last 5 and 15 minutes, which doesn't make any sense with that little traffic.

Just a thought.

File: container/root/etc/php5/fpm/conf.d/web.user.ini [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 memory_limit = 192M
 max_execution_time = 60
+request_terminate_timeout=60
```
Comment (mannytoledo) [inline container/root/etc/php5/fpm/conf.d/web.user.ini:6]: For now this is probably fine but we may want to chop down both the `max_execution_time` and `request_terminate_timeout` in the future. As we discussed, the clock starts from the client request and if we keep these all at a minute it degrades the client experience.

**Comment by mannytoledo** [comment]
:+1:

**Comment by chrishenry** [comment]
Other than maybe tweaking some other timeout params, LGTM

**Comment by jimdelois** [comment]
:white_check_mark:

**Pull request merged**